### PR TITLE
Fixes 52: Add EditContent Modal

### PR DESCRIPTION
**Pull request by Andrewgdewar**
This PR adds the ability to Edit repositories within a list. 

Bulk edit functionality is also complete within the modal should we need it.
We would just need the multi-select boxes/Bulk Edit button/BulkEditAPI, to see that unrequested "feature" complete. 😜 

Additional Changes: 
- Introduced new "single item" UI for both AddContent and EditContent (more streamlined).
- Fixed a bug associated with the textfield on the "name/url" field on filters not being cleared properly
- Fixed a bug associated with the versions not displaying/being stored correctly.
- Fixed a bug associated with AddContent Modal validity after removing items.

### CODE REVIEW

**Comment by jlsherrill** [comment]
https://issues.redhat.com/browse/HMSCONTENT-52

File: src/components/EditContentModal/EditContentModal.tsx [hunk 1]
```diff
@@ -0,0 +1,445 @@
...
+  };
+
+  const urlOnBlur = (index: number) => {
+    updateArchAndVersion(index);
+  };
+
+  return (
+    <Modal
+      variant={ModalVariant.medium}
+      title='Edit Custom Content'
```
Comment (rverdile) [inline src/components/EditContentModal/EditContentModal.tsx:205]: This says "Custom Content", everything else says "Content Source". Is that intentional?

**Review by rverdile — COMMENTED** [review]
If I snap my window to the side of my screen, the modal closes. Is that working as intended?

File: src/components/EditContentModal/EditContentModal.tsx (hunk 1 above)
Comment (Andrewgdewar) [reply src/components/EditContentModal/EditContentModal.tsx:205]: nope

**Comment by Andrewgdewar** [comment]
> If I snap my window to the side of my screen, the modal closes. Is that working as intended?


Unfortunately, the entire page gets remounted when you do that. This is due to the parent switching from a mobile/desktop wrapper. There's a specific breakpoint (1200px) that this happens at, I'm not sure if there's a workaround.

**Review by rverdile — APPROVED** [review]
lgtm functionally. I can't really speak to the code. I need to familiarize myself more with formik.

File: src/services/Content/ContentApi.ts [hunk 2]
```diff
@@ -104,6 +115,16 @@ export const AddContentListItems: (request: CreateContentRequest) => Promise<voi
   return data;
 };
 
+export const EditContentListItem: (request: EditContentRequestItem) => Promise<void> = async (
+  request,
+) => {
+  const { data } = await axios.patch(
+    `/api/content_sources/v1.0/repositories/${request.uuid}`,
```
Comment (mshriver) [inline src/services/Content/ContentApi.ts:122]: This needs changed to content-sources, see #30

**Pull request merged**